image/docker_schema1.go

```diff
@@ -343,8 +354,8 @@ func (m *manifestSchema1) convertToManifestSchema2(uploadedLayerInfos []types.Bl
 		Digest:    digest.FromBytes(configJSON),
 	}
 
-	m2 := manifestSchema2FromComponents(configDescriptor, nil, configJSON, layers)
-	return memoryImageFromManifest(m2), nil
+	m2 := manifestSchema2FromComponents(configDescriptor, nil, configJSON, layers).(*manifestSchema2)
```

Comment:
Redeclare `manifestSchema2FromComponents` to return a `*manifestSchema2` explicitly instead of making a runtime cast here.

Reply:
… actually, is either necessary? AFAICT both `memoryImageFromManifest` and `.UpdatedImage()` can work on the `genericManifset` interface.

Reply:
you're right, I'm dumb